presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceDictionarySelectiveReader.java

```diff
@@ -288,9 +270,9 @@ private int readWithFilter(int[] positions, int positionCount)
             }
             streamPosition++;
 
-            if (filter != null) {
-                outputPositionCount -= filter.getPrecedingPositionsToFail();
-                int succeedingPositionsToFail = filter.getSucceedingPositionsToFail();
+            if (context.getFilter() != null) {
```

Comment:
Adding a `context.hasFilter()` for `context.getFilter() != null` will make it a bit better

Reply:
In the next PR, I will change this.